scene/main/scene_tree.cpp

```diff
@@ -1366,6 +1366,8 @@ SceneTree::SceneTree() {
 	root_lock = 0;
 	node_count = 0;
 
+	Math::randomize();
```

Comment:
I'd call this somewhere at `register_core_types()` to randomize this earlier. But this is technically more local and closer to project, so I have no strong opinion about this.

C++ custom modules don't have such low-level access to the scene tree initialization, and the only way to randomize custom RNG implementations is through `register_modules_types()` and via constructors, so I'd prefer we have some consistency in behavior.

Reply:
Well, I could put it in the constructor of Math class, but it's empty and all the methods are static, so I thought it might not be a good idea.

Reply:
Yeah I think it's more like for scoping methods together without using `namespace`s in Godot, like in `Geometry2D`.

Reply:
I think it's fine for this to be randomized at `SceneTree` init, because the main issue is confusion beginners have with having to call `randomize()`, for engine stuff (like tests) you typically set the seed manually anyways, this is more to simplify gameplay programming stuff for beginners (and those who are annoyed by the fact that `randomize()` should be called manually, of course).